cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -766,17 +797,14 @@ _CCCL_DEVICE void transform_kernel_ublkcp(
       {
         char* smem         = smem_base;
         auto fetch_operand = [&](auto aligned_ptr) {
-          using T         = typename decltype(aligned_ptr)::value_type;
-          const char* src = smem;
-          if constexpr (alignof(T) < bulk_copy_alignment)
-          {
-            src += aligned_ptr.head_padding;
-          }
-          else
+          using T                = typename decltype(aligned_ptr)::value_type;
+          const int head_padding = alignof(T) < bulk_copy_alignment ? aligned_ptr.head_padding : 0;
+          if constexpr (!tile_sizes_retain_max_alignment)
           {
-            _CCCL_ASSERT(aligned_ptr.head_padding == 0, "");
+            smem = round_up_smem_ptr<alignof(T)>(smem);
           }
-          smem += int{sizeof(T)} * tile_size + bulk_copy_alignment;
+          const char* src = smem + head_padding;
```

Comment:
Would it make sense to annotate here with `_CCCL_BUILTIN_ASSUME_ALIGNED?`

Reply:
I tried and it didn't make a difference.